salalib/axialmap.cpp

```diff
@@ -4171,30 +4171,26 @@ bool ShapeGraph::angularstepdepth(Communicator *comm)
 
 // helper -- a little class to tidy up a set of lines
 
-void TidyLines::tidy(prefvec<Line>& lines, const QtRegion& region)
+void TidyLines::tidy(std::vector<Line>& lines, const QtRegion& region)
 {
-   size_t i = 0;
    m_region = region;
    double maxdim = __max(m_region.width(),m_region.height());
 
    // simple first pass -- remove very short lines
-   pvecint removelist;
-   for (i = 0; i < lines.size(); i++) {
-      if (lines[i].length() < maxdim * TOLERANCE_B) {
-         removelist.add(i);
-      }
-   }
-   lines.remove_at(removelist);
-   removelist.clear();  // always clear this list, it's reused
+   lines.erase(
+               std::remove_if(lines.begin(), lines.end(),
+                              [maxdim](const Line& line)
+   {return line.length() < maxdim * TOLERANCE_B;}), lines.end());
 
    // now load up m_lines...
    initLines(lines.size(),m_region.bottom_left,m_region.top_right);
-   for (i = 0; i < lines.size(); i++) {
-      addLine(lines[i]);
+   for (auto& line: lines) {
+      addLine(line);
    }
    sortPixelLines();
 
-   for (i = 0; i < lines.size(); i++) {
+   std::set<int> removelist;

```

Comment:
Any particular reason for using a `std::set` here? Wouldn't a vector do as well as the indices come in in order anyway?